073-improve-handling-of-CA-renewals-and-replacements-in-client-based-operands.md

```diff
@@ -0,0 +1,231 @@
+   At this point in time, the Kafka brokers still use the old server certificates signed by the old CA.
+3. The Kafka client based operand (e.g. Kafka Connect cluster) using the typical `.spec.tls` configuration:
+   ```yaml
+   tls:
+      trustedCertificates:
+         - secretName: my-cluster-cluster-ca-cert
+           certificate: ca.crt
+   ```
+   detects that the `ca.crt` in the Secret was updated and triggers a rolling update of the operand to use the new CA.
+   After the rolling update, it ceases to trust the old CA still in use by the Kafka brokers, resulting in the operand failing to connect to the brokers.
+4. After all brokers are rolled to trust the new CA, another rolling update happens and new server certificates signed by the new CA are rolled out to the Kafka brokers in another rolling update.
+5. Once all brokers are rolled once again and use the new CA, the operand such as Kafka Connect will be able to connect again and start working.
+   The impact and length of the outage will depend on many different aspects:
+   * How many nodes does the operand such as Kafka Connect have.
+     If it is only one replica, the whole operand will be down.
+     With multiple replicas, only one replica should be down.
+   * How big and busy the Kafka cluster is.
+     A small cluster might take few minutes to move to the new CA completely.
+     A big cluster with many nodes that are busy and need more time to re-sync between restarts might take much longer.
+6. After all brokers (and other components such as ZooKeeper, Topic and User Operator, Cruise Control etc.) use the new CA, another rolling update will be done to completely remove the trust to the old CA by removing the earlier `ca-<timestamp>.crt` from the Secret and rolling all components once again.
+   This last rolling update does not impact operands such as Kafka Connect.
+
+## Proposal
+
+This proposal suggests improving the API of our custom resources that we use for configuring the trusted certificates.
+It should allow loading more certificates without an exact specification of the keys in the Secret.
+To do so, we should add a new field called `certificatesMatching`.
+This field will allow users to specify a pattern to match the files that should be loaded from the Secret without specifying the exact name.
+The pattern will use the [_glob_ pattern format](https://en.wikipedia.org/wiki/Glob_(programming)) allowing users to use wildcards such as `*` or `?` to specify the certificates that should be loaded.
+
+In the Strimzi Cluster Operator, the [Java `PathMatcher`](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/nio/file/PathMatcher.html) will be used to parse the pattern.
+In the container, `bash` will be used to evaluate the pattern.
+
+The existing `certificate` field will remain supported.
+But it will not be marked as required anymore and instead the CRDs will require one of the fields `certifciate` or `certificatesMatching` to be specified.
```

Comment:
```suggestion
But it will not be marked as required anymore and instead the CRDs will require one of the fields `certificate` or `certificatesMatching` to be specified.
```